tcp.go

```diff
@@ -95,13 +95,21 @@ func DisableReuseport() Option {
 		return nil
 	}
 }
+
 func WithConnectionTimeout(d time.Duration) Option {
 	return func(tr *TcpTransport) error {
 		tr.connectTimeout = d
 		return nil
 	}
 }
 
+func WithResourceManager(rcmgr network.ResourceManager) Option {
```

Comment:
We might want to add this as a parameter to the `New` function instead of an option. Then the go-libp2p constructor magic can figure out the parameters.

Reply:
yeah, agreed.